core/src/main/java/com/digitalpebble/stormcrawler/util/URLPartitioner.java

```diff
@@ -14,38 +14,37 @@
  */
 package com.digitalpebble.stormcrawler.util;
 
-import com.digitalpebble.stormcrawler.Constants;
 import com.digitalpebble.stormcrawler.Metadata;
-import crawlercommons.domains.PaidLevelDomain;
-import java.net.InetAddress;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.Map;
 import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Generates a partition key for a given URL based on the hostname, domain or IP address. This can
  * be called by the URLPartitionerBolt or any other component.
  */
-public class URLPartitioner {
+public final class URLPartitioner {
```

Comment:
Any reason for being `final` here? Might break existing code bases for people extending this class.

Reply:
Nope no reason, I was just overzealous again.
I'll change that back.

Reply:
After looking at it again: 
I don't think that opening the `URLPartitioner ` makes a lot of sense because all the usages in the existing code are direct constructor calls and `protected final` and `private` fields.
And neither the old code nor the new code are meant to be overridden because the initialized `mode`-field is/was set to `private`.

I also found no method in SC that consumes a URLPartitioner, if people inherit from it they should initialize it via the Constructor and `configure`.